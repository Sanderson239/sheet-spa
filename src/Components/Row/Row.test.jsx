import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Row } from './Row';

describe('Spreadsheet Row', () => {
  it('should render a Row with rowId and rowsById', () => {
    const rowId = 1;
    const rowsById = { 1: { cellIds: [1, 2, 3, 4, 5], cellsById: { 1: 'A2', 2: 'B2', 3: 'C2', 4: 'D2', 5: 'E2' } } };
    const component = shallow(
      <Row
        rowId={rowId}
        rowsById={rowsById}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
