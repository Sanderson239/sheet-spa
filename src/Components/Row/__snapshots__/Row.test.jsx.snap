// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spreadsheet Row should render a Row with rowId and rowsById 1`] = `
<TableRow
  as="tr"
  cellAs="td"
>
  <Component
    cellId={1}
    key="1"
    rowId={1}
  />
  <Component
    cellId={2}
    key="2"
    rowId={1}
  />
  <Component
    cellId={3}
    key="3"
    rowId={1}
  />
  <Component
    cellId={4}
    key="4"
    rowId={1}
  />
  <Component
    cellId={5}
    key="5"
    rowId={1}
  />
</TableRow>
`;
